### PR TITLE
Washing machines wiggle while running, can be unanchored

### PR DESCRIPTION
**Pull request by SpaceManiac**
:cl:
add: Washing machines now wiggle slightly while running.
add: Washing machines can be unanchored if their panel is open for even more wiggle.
/:cl:

washing-1

washing-2

### CODE REVIEW

**Comment by Grazyn** [comment]
make it so you can ~~sit~~ climb on them

**Comment by MrDoomBringer** [comment]
@Grazyn oh my god yes. Make it have a chance to throw you off too!!
image

File: code/game/machinery/washing_machine.dm [hunk 1]
```diff
@@ -39,8 +39,27 @@
...
+			M.Translate(rand(-1, 1), rand(0, 1))
+			animate(src, transform=M, time=1)
+			animate(transform=matrix(), time=1)
+	else
+		if (prob(1))
+			step(src, pick(GLOB.cardinals))
+		var/matrix/M = new
+		M.Translate(rand(-3, 3), rand(-1, 3))
+		animate(src, transform=M, time=2)
+	addtimer(CALLBACK(src, .proc/wiggle), 2)
```
Comment (ninjanomnom) [inline code/game/machinery/washing_machine.dm]: Use a cancelable looping timer for this sort of thing

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
Is your washing machine running?

...

...

...

Better go catch it!!

**Comment by Kagus** [comment]
Let us add sandstone bricks to the wash cycle in order to detonate the entirety of dorms.

File: code/game/machinery/washing_machine.dm (hunk 1 above)
Comment (SpaceManiac) [reply code/game/machinery/washing_machine.dm]: How would that compare to using `SSfastprocess`, which has the appropriate `wait` of 2?

Comment (ninjanomnom) [reply code/game/machinery/washing_machine.dm]: If you have a subsystem you can use that would be better

**Comment by SpaceManiac** [comment]
Changed to use `SSfastprocess`, will require #39322 in order to work correctly

**Pull request merged**